### ISSUE TITLE
Cannot Cast DummyRepository to GitRepository on tasks

### ISSUE THREAD

**Issue by totahuanocotl** [allegro/axion-release-plugin#186]
Hi, we have a scenario where we can't clone a git repository, so we download the archived sources and run some of our build tasks, which don't involve any versioning, thought the cV task is called (but it's value is not used).

We tried to upgrade from version `1.3.3` to the latest `1.7.0` but it seems that the behaviour of the plugin has changed, from logging a warning message, to failing the build with a `ClassCastException`.

I am aware of https://github.com/allegro/axion-release-plugin/issues/133, but i think this is a different case, as we are not trying to do a release, rather just executing unrelated tasks. 

We would have expected the same behaviour on all the versions. So far we don't require any of the changes on the new versions, but unfortunately this will prevent us from upgrading to any new versions if the behaviour is kept as it currently stands.   

Here is the output for the different versions I've tried, and you can see it started failing from `1.5.0`.

For `1.3.3` version the behaviour is:
```
$ ./gradlew cV --refresh-dependencies
Failed top open repository, trying to work without it org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: ~/workspace/no-git
Project version: -SNAPSHOT
BUILD SUCCESSFUL
```

For `1.4.1`:
```
$ ./gradlew cV --refresh-dependencies
Failed to open repository, trying to work without it org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: ~/workspace/no-git
Could not resolve current position on uninitialized repository, returning default

Project version: -SNAPSHOT

BUILD SUCCESSFUL
```

For `1.5.0`, `1.6.0`, `1.7.0`:
```
$ ./gradlew cV --refresh-dependencies
Failed to open repository, trying to work without it org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: ~/workspace/no-git

FAILURE: Build failed with an exception.

* Where:
Build file '~/workspace/no-git/myProject/build.gradle' line: 15

* What went wrong:
A problem occurred evaluating project ':myProject'.
> Cannot cast object 'pl.allegro.tech.build.axion.release.infrastructure.DummyRepository@345ade19' with class 'pl.allegro.tech.build.axion.release.infrastructure.DummyRepository' to class 'pl.allegro.tech.build.axion.release.infrastructure.git.GitRepository'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

**Comment by adamdubiel** [comment]
Looks like a regression bug - it should work without git repo. Will fix it.

**Comment by totahuanocotl** [comment]
thanks @adamdubiel!

**Comment by adamdubiel** [comment]
@totahuanocotl released as 1.7.1
